core/dbt/clients/registry.py

```diff
@@ -34,9 +34,33 @@ def _get_url(name, registry_base_url=None):
     return "{}{}".format(registry_base_url, url)
 
 
-def _get_with_retries(package_name, registry_base_url=None):
+def _check_package_redirect(package_name: str, response: Dict[str, Any]) -> None:
+    # Either redirectnamespace or redirectname in the JSON response indicate a redirect
+    # redirectnamespace redirects based on package ownership
+    # redirectname redirects based on package name
+    # Both can be present at the same time, or neither. Fails gracefully to old name
+    if ("redirectnamespace" in response) or ("redirectname" in response):
+        if ("redirectnamespace" in response) and response["redirectnamespace"] is not None:
+            use_namespace = response["redirectnamespace"]
+        else:
+            use_namespace = response["namespace"]
+
+        if ("redirectname" in response) and response["redirectname"] is not None:
+            use_name = response["redirectname"]
+        else:
+            use_name = response["name"]
```

Comment:
Is there a reason not to use a get here instead of the nested ifs?
```suggestion
        use_namespace = response.get("redirectnamespace") or response["namespace"]
        use_name = response.get("redirectname") or response["name"]
```

Reply:
I don't think there is a reason not to. I had just shifted the existing logic from the `package` method into it's own method (and instead called from `_get_package_with_retries`. Thank you for pointing out this possible refactor, it will make it nicer to read/maintain. I'll get it updated, the logic looks equivalent 🚀